### PR TITLE
DiscoPoP Wrappers

### PR DESCRIPTION
**Pull request by lukasrothenberger**
TODOs:
- [x] remove hardcoded paths

### CODE REVIEW

File: DiscoPoP/DiscoPoP.cpp [hunk 1]
```diff
@@ -904,7 +904,9 @@ void DiscoPoP::instrument_module(llvm::Module *module, map <string, string> *tru
             inlinedFunction(func)) {
             continue;
         }
+        errs() << "Instrumenting function\n";
```
Comment (lukasrothenberger) [inline DiscoPoP/DiscoPoP.cpp:907]:
```suggestion
```

File: DiscoPoP/DiscoPoP.cpp [hunk 2]
```diff
@@ -904,7 +904,9 @@ void DiscoPoP::instrument_module(llvm::Module *module, map <string, string> *tru
...
         instrument_function(func, trueVarNamesFromMetadataMap);
+        errs() << "\tInstrumenting function done.\n";
```
Comment (lukasrothenberger) [inline DiscoPoP/DiscoPoP.cpp:909]:
```suggestion
```

File: DiscoPoP/DiscoPoP.cpp [hunk 3]
```diff
@@ -934,7 +936,9 @@ void DiscoPoP::instrument_function(llvm::Function *function, map <string, string
 
     for (auto loop_it = loop_info.begin(); loop_it != loop_info.end();
          ++loop_it) {
+        errs() << "Instrumenting loop\n";
```
Comment (lukasrothenberger) [inline DiscoPoP/DiscoPoP.cpp:939]:
```suggestion
```

File: DiscoPoP/DiscoPoP.cpp [hunk 4]
```diff
@@ -934,7 +936,9 @@ void DiscoPoP::instrument_function(llvm::Function *function, map <string, string
...
         instrument_loop(*function, file_id, *loop_it, loop_info, trueVarNamesFromMetadataMap);
+        errs() << "Instrumenting loop done\n";
```
Comment (lukasrothenberger) [inline DiscoPoP/DiscoPoP.cpp:941]:
```suggestion
```

File: DiscoPoP/DiscoPoP.cpp [hunk 5]
```diff
@@ -1130,14 +1131,15 @@ void DiscoPoP::instrument_loop(Function &F, int file_id, llvm::Loop *loop, LoopI
                     if (bbName.find("if") != std::string::npos ||
                         bbName.find("for") != std::string::npos) {
                         // e.g. in lulesh.cc: "if (domain.vdov(indx) != Real_t(0.)) { if ( dtf < dtcourant_tmp ) { dtcourant_tmp = dtf ; courant_elem  = indx ; }}"
+                        check_value_usage(candidate.store_inst_->getValueOperand(), cast<Value>(candidate.load_inst_));
```
Comment (lukasrothenberger) [inline DiscoPoP/DiscoPoP.cpp:1134]:
```suggestion
```

File: DiscoPoP/DiscoPoP.cpp [hunk 6]
```diff
@@ -1990,27 +1998,31 @@ bool DiscoPoP::runOnModule(Module &M) {
         runOnFunction(F);
     }
 
-    //cout << "\n\tFunctions Done.\n";
+    errs() << "\n\tFunctions Done.\n";
```
Comment (lukasrothenberger) [inline DiscoPoP/DiscoPoP.cpp:2001]:
```suggestion
```

File: DiscoPoP/DiscoPoP.cpp [hunk 7]
```diff
@@ -1990,27 +1998,31 @@ bool DiscoPoP::runOnModule(Module &M) {
...
 
     // DPReduction
     module_ = &M;
     ctx_ = &module_->getContext();
+    errs() << "Retrieved context\n";
```
Comment (lukasrothenberger) [inline DiscoPoP/DiscoPoP.cpp:2006]:
```suggestion
```

File: DiscoPoP/DiscoPoP.cpp [hunk 8]
```diff
@@ -1990,27 +1998,31 @@ bool DiscoPoP::runOnModule(Module &M) {
...
 
     reduction_file = new std::ofstream();
     reduction_file->open("reduction.txt", std::ios_base::app);
 
     loop_counter_file = new std::ofstream();
     loop_counter_file->open("loop_counter_output.txt", std::ios_base::app);
+    errs() << "Opened file\n";
```
Comment (lukasrothenberger) [inline DiscoPoP/DiscoPoP.cpp:2013]:
```suggestion
```

File: DiscoPoP/DiscoPoP.cpp [hunk 9]
```diff
@@ -1990,27 +1998,31 @@ bool DiscoPoP::runOnModule(Module &M) {
...
 
     bool success = dp_reduction_init_util(FileMappingPath);
     if (!success) {
         llvm::errs() << "could not find the FileMapping file: " << FileMappingPath << "\n";
         return false;
     }
-
+    errs() << "Instrumenting module\n";
```
Comment (lukasrothenberger) [inline DiscoPoP/DiscoPoP.cpp:2020]:
```suggestion
```

File: DiscoPoP/DiscoPoP.cpp [hunk 10]
```diff
@@ -1990,27 +1998,31 @@ bool DiscoPoP::runOnModule(Module &M) {
...
     instrument_module(&M, &trueVarNamesFromMetadataMap);
+    errs() << "Instrumentation done\n";
 
     dp_reduction_insert_functions();
+    errs() << "Inserted functions\n";
```
Comment (lukasrothenberger) [inline DiscoPoP/DiscoPoP.cpp:2025]:
```suggestion
```

File: DiscoPoP/DiscoPoP.cpp [hunk 11]
```diff
@@ -1990,27 +1998,31 @@ bool DiscoPoP::runOnModule(Module &M) {
...
 
     bool success = dp_reduction_init_util(FileMappingPath);
     if (!success) {
         llvm::errs() << "could not find the FileMapping file: " << FileMappingPath << "\n";
         return false;
     }
-
+    errs() << "Instrumenting module\n";
     instrument_module(&M, &trueVarNamesFromMetadataMap);
+    errs() << "Instrumentation done\n";
```
Comment (lukasrothenberger) [inline DiscoPoP/DiscoPoP.cpp:2022]:
```suggestion
```

File: DiscoPoP/DiscoPoP.cpp [hunk 12]
```diff
@@ -2021,6 +2033,7 @@ bool DiscoPoP::runOnModule(Module &M) {
         loop_counter_file->flush();
         loop_counter_file->close();
     }
+    errs() << "Written files\n";
```
Comment (lukasrothenberger) [inline DiscoPoP/DiscoPoP.cpp:2036]:
```suggestion
```

File: DiscoPoP/DiscoPoP.cpp [hunk 13]
```diff
@@ -2069,6 +2082,7 @@ bool DiscoPoP::runOnFunction(Function &F) {
 
     // CUGeneration
     {
+        errs() << "GU GEN START\n";
```
Comment (lukasrothenberger) [inline DiscoPoP/DiscoPoP.cpp:2085]:
```suggestion
```

File: DiscoPoP/DiscoPoP.cpp [hunk 14]
```diff
@@ -2441,6 +2459,7 @@ bool DiscoPoP::runOnFunction(Function &F) {
         staticDependencyFile->close();
 
         if (DP_hybrid_DEBUG) errs() << "Done with function " << F.getName() << ":\n";
+        errs() << "DP INST OMISSION END\n";
```
Comment (lukasrothenberger) [inline DiscoPoP/DiscoPoP.cpp:2462]:
```suggestion
```

**Comment by goerlibe** [comment]
- [ ]  hardcoded compiler names

**Review by goerlibe — APPROVED** [review]

**Pull request merged**